### PR TITLE
Use electron.net on lastfm requests

### PR DESCRIPTION
**Pull request by asermax**
Fixes related to #921
This change monkey-patches the external `lastfm-node` library to use `electron.net` instead of node's `http` module.

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
@asermax Can you confirm if this works or not? 👍

File: src/main/features/core/lastFM.js [hunk 1]
```diff
@@ -1,9 +1,30 @@
 import { BrowserWindow } from 'electron';
-import { LastFmNode } from 'lastfm';
 import path from 'path';
 
 import { LASTFM_API_KEY, LASTFM_API_SECRET } from '../../constants';
 
+// hijack lastfm module to use 'electron.net' instead of 'http'
+global.GENTLY_HIJACK = {
+  hijack(require) {
+    return function (moduleName) {
```
Comment (MarshallOfSound) [inline src/main/features/core/lastFM.js]: `function` requires a name 👍

**Comment by asermax** [comment]
@MarshallOfSound I did a test it earlier today but with no success. Not being able to debug is a real pain too :( I didn't clean the previous installation when testing it tho, so I'll completely remove the player from my comp and give it another try tomorrow just in case and get back to you.
Beyond the eslint issue, does this change look ok? Can you think of any reason why it could be failing?

File: src/main/features/core/lastFM.js (hunk 1 above)
Comment (asermax) [inline src/main/features/core/lastFM.js]: I'll change it for an arrow function since it seems that's the norm on the code (I didn't realize earlier 🙇)

**Comment by asermax** [comment]
Tried again with no success :(

**Comment by rattboi** [comment]
@asermax Any update on this issue? Just started a new job with http proxy and running into this again, lol.

**Comment by asermax** [comment]
@rattboi not really, I did some tests with no success and came back to using the web app with a chrome plugin for lastfm :<

**Comment by MarshallOfSound** [comment]
Closing due to lack of activity, if you still wish this PR to go through please rebase onto current master and follow up with the provided comments

**Pull request closed without merging**